app/stores/base/Store.ts

```diff
@@ -144,6 +149,42 @@ export default abstract class Store<T extends Model> {
     LifecycleManager.executeHooks(model.constructor, "afterRemove", model);
   }
 
+  @action
+  addToArchive(item: PartialWithArchivedAt<T>): void {
+    const inverseRelations = getInverseRelationsForModelClass(this.model);
+
+    inverseRelations.forEach((relation) => {
+      const store = this.rootStore.getStoreForModelName(relation.modelName);
+      if ("orderedData" in store) {
+        const items = (store.orderedData as Model[]).filter(
+          (data) => data[relation.idKey] === item.id
+        );
+
+        items.forEach((item) => {
+          let archiveBehavior = relation.options.onArchive;
+
+          if (typeof relation.options.onArchive === "function") {
+            archiveBehavior = relation.options.onArchive(item);
+          }
+
+          if (archiveBehavior === "cascade") {
+            store.addToArchive(item as any);
+          } else if (archiveBehavior === "null") {
+            item[relation.idKey] = null;
+          }
+        });
+      }
+    });
+
+    // Remove associated policies automatically, not defined through Relation decorator.
+    if (this.modelName !== "Policy") {
+      this.rootStore.policies.remove(item.id);
+    }
+
+    item.archivedAt = new Date().toISOString();
```

Comment:
If item has `archivedAt` why are we resetting it here?

Reply:
Not at this point. The items would have been archived on server but that info has not _yet_ propagated to all concerned clients yet.

Reply:
The type suggests it exists, I suppose this is just to make TS happy? Perhaps also solved by ArchivableModel then

Reply:
Yeah, the type there is just a "hack" now to make it work. Had `ArchivableModel` in mind actually.

Reply:
Also, the type suggests that it has the `archivedAt` attribute, _not_ that it might be `null | undefined`.